### PR TITLE
PR: Implement support for "OpenColorIO" processor.

### PR DESCRIPTION
**Pull request by KelSolaar**
References #654.

The API is as follows:

```python
>>> config = os.path.join(os.path.dirname(__file__), 'tests', 'resources',  'config-aces-reference.ocio.yaml')
>>> a = full([4, 2, 3], 0.18)
>>> process_image_OpenColorIO(a, 'ACES - ACES2065-1', 'ACES - ACEScct', config=config)
array([[[ 0.4135878...,  0.4135878...,  0.4135878...],
        [ 0.4135878...,  0.4135878...,  0.4135878...]],

       [[ 0.4135878...,  0.4135878...,  0.4135878...],
        [ 0.4135878...,  0.4135878...,  0.4135878...]],

       [[ 0.4135878...,  0.4135878...,  0.4135878...],
        [ 0.4135878...,  0.4135878...,  0.4135878...]],

       [[ 0.4135878...,  0.4135878...,  0.4135878...],
        [ 0.4135878...,  0.4135878...,  0.4135878...]]], dtype=float32)
>>> process_image_OpenColorIO(a, 'ACES - ACES2065-1', 'Display - sRGB',  'Output - SDR Video - ACES 1.0', ocio.TRANSFORM_DIR_FORWARD, config=config)
array([[[ 0.3559523...,  0.3559525...,  0.3559525...],
        [ 0.3559523...,  0.3559525...,  0.3559525...]],

       [[ 0.3559523...,  0.3559525...,  0.3559525...],
        [ 0.3559523...,  0.3559525...,  0.3559525...]],

       [[ 0.3559523...,  0.3559525...,  0.3559525...],
        [ 0.3559523...,  0.3559525...,  0.3559525...]],

       [[ 0.3559523...,  0.3559525...,  0.3559525...],
        [ 0.3559523...,  0.3559525...,  0.3559525...]]], dtype=float32)
```

it is also possible to use the *OCIO* config from the environment.

### CODE REVIEW

File: colour/io/ocio.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    config = (ocio.Config.CreateFromEnv()
+              if config is None else ocio.Config.CreateFromFile(config))
+
+    a = np.atleast_3d(as_float_array(a, dtype=np.float32))
+    height, width, channels = a.shape
+
+    processor = config.getProcessor(*args)
+
+    processor = (processor.getDefaultCPUProcessor()
+                 if use_cpu else processor.getDefaultGPUProcessor())
```
Comment (zachlewis) [inline colour/io/ocio.py:104]: I don't think you can use a GPU Processor like you can a CPU Processor; i.e., there's no "apply" method....

Comment (KelSolaar) [reply colour/io/ocio.py:104]: I actually haven't tried nor looked at it! I will dig through the OCIO codebase.

Comment (KelSolaar) [reply colour/io/ocio.py:104]: So looks like it is more involved indeed: https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/0021372b4dba3444e21a36d846b327efad054f8d/tests/python/GpuShaderDescTest.py#L127

Comment (remia) [reply colour/io/ocio.py:104]: And on top of that you need to create an OpenGL context, manually upload any texture data needed and execute your draw call...!

Comment (KelSolaar) [reply colour/io/ocio.py:104]: Yeah, I saw that, will try to update that tonight, be great to consider making that easier, there is probably no good reasons why it cannot be abstracted for a user, c.f. `import cupy as np`.

Comment (KelSolaar) [reply colour/io/ocio.py]: I went path-of-least-resistance and removed the non-working GPU path, we can figure that out in a later PR.

Comment (KelSolaar) [reply colour/io/ocio.py]: So, confirming that we will not have GPU processing anytime soon: https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1493/commits/4a8dac4affef4fdea3c2c5b4d67abde7c803c645

**Comment by KelSolaar** [comment]
Merging this one!

**Pull request merged**